gst/nnstreamer/tensor_repo.c

```diff
@@ -239,31 +310,41 @@ gst_tensor_repo_set_eos (guint nth)
  * @brief get GstTensorRepoData from repo
  */
 GstBuffer *
-gst_tensor_repo_get_buffer (guint nth)
+gst_tensor_repo_get_buffer (guint nth, guint o_nth, gboolean * eos,
+    guint * newid)
 {
   GstTensorRepoData *current_data;
   GstBuffer *buf;
-  gboolean eos = FALSE;
+  gboolean slotid_changed = FALSE;
 
   GST_TENSOR_REPO_LOCK (nth);
   current_data = gst_tensor_repo_get_repodata (nth);
 
   while (!current_data->buffer) {
+    if (gst_tensor_repo_check_changed (nth, newid, FALSE)) {
+      slotid_changed = TRUE;
+      buf = NULL;
+      break;
+    }
+
     if (gst_tensor_repo_check_eos (nth)) {
-      eos = TRUE;
+      *eos = TRUE;
       break;
     }
     GST_TENSOR_REPO_WAIT_PUSH (nth);
   }
-  if (eos) {
-    if (DBG)
-      GST_DEBUG ("Get EOS Signal while waiting\n");
-    buf = NULL;
-  } else {
-    buf = gst_buffer_copy_deep (current_data->buffer);
-    if (DBG) {
-      unsigned long size = gst_buffer_get_size (buf);
-      GST_DEBUG ("Popped [ %d ] (size: %lu)\n", nth, size);
+
+  if (!slotid_changed) {
+    if (*eos) {
+      if (DBG)
+        GST_DEBUG ("Get EOS Signal while waiting\n");
+      buf = NULL;
+    } else {
+      buf = gst_buffer_copy_deep (current_data->buffer);
+      if (DBG) {
+        unsigned long size = gst_buffer_get_size (buf);
+        GST_DEBUG ("Popped [ %d ] (size: %lu)\n", nth, size);
+      }
     }
   }
   current_data->buffer = NULL;
```

Comment:
When called gst_tensor_repo_set_buffer(), 
```
data->buffer = gst_buffer_copy (buffer);
```

I think that it needs gst_buffer_unref (current_data->buffer) before set current_data->buffer to null.